docs/how-to-maintain.rst

```diff
@@ -0,0 +1,113 @@
+* ``patch`` is incremented when you make backwards compatible bug fixes.
+
+You can add a suffix to signal more information on the current version.
+
+* Use ``0.0.1dev0`` to signal that the release is a development version which is
+  currently tested and to be released soon. Increment the last digit for multiple
+  development versions.
+
+.. note::
+
+   The development suffix is not separated with a dot from the patch version as this
+   does not comply with the publishing process on anaconda.org.*
+
+
+Branching Model
+---------------
+
+The branching model for GETTSIM is very simple.
+
+1. The master branch is the target of ongoing development. Every PR should be merged
+   into master.
+
+   (The only deviation can happen with respect to backports. See :ref:`backports` for
+   more.)
+
+2. There can exist branches for minor versions called, for example, ``0.1.``. The
+   branches contain changes, e.g. backports, which are not released yet. To find out how
+   to release the next patch version, see the following section. If there are no
+   additional changes waiting to be released, the branches should be deleted for
+   clarity.
+
+
+.. _releases:
+
+How To Release
+--------------
+
+To release a new version of GETTSIM, do the following.
+
+1. To start a new release for version ``0.2.1``, create a new milestone with the version
+   as its name on Github to collect issues and PRs which should be included in the new
+   version.
+
+2. Call a halt on changes to the master branch. Make sure all changes to the master
+   branch since the last version are included in the milestone.
+
+**Here, we have to decide whether we allows pushing to master for releases or not. Then,
+we can use bump2version to create a tagged commit. Otherwise, we stick to PRs and create
+the tag via Github. I would go with the latter.**
+
+3. Create a PR to finalize the new release.
+
+   1. Update ``CHANGES.rst`` with all necessary information regarding the new release.
+   2. Use ``bump2version [major|minor|patch]`` to increment all version strings.
+   3. Merge it.
+
+4. Go to the `page for releases <https://github.com/iza-institute-of-labor-economics/
+   gettsim/releases>`_ and draft a new release. The tag and title become ``v0.2.1``.
+   Make sure to target the master branch. A description is not necessary as the most
+   important information lies in ``CHANGES.rst``.
+
+   Release the new version by clicking "Publish release".
+
+
+.. _backports:
+
+How To Backport
+---------------
+
+Backporting is the process of applying a change in future versions of GETTSIM to older
+versions
+
+Scope
+^^^^^
+
+As backports can introduce new regressions the scope is limited to critical bug fixes
+and documentation changes. Performance enhancements and new features are not backported.
+
+Procedure
+^^^^^^^^^
+
+In the following we will consider an example where GETTSIM stable version is 0.2.0.
+Version 0.2.1 is currently developed on the master branch but not released. And a bug
```

Comment:
As I understood the model, if `0.2.0` is released, we have a branch `0.2` which takes backports and we develop 0.3 on master?

If correct, we should take out one level here and backport to 0.2.1, forgetting about 0.1.8 but noting that we may even want to backport to multiple minor versions.

Reply:
I agree that we would develop `0.3` and not `0.2.1` on master, so we would have to backport to `0.2.1`, `0.1.9` and potentially some `0.0.x`. 

I hadn't really thought about backporting before, but have a few general questions:

- In the described workflow we never just delete a release. This means that there will be versions with known bug on our conda channel. Is this what we want? I'm thinking about a bug that leads to wrong results without warnings or exceptions. 
- Maybe we should suggest that people never specify the third digit in their environment files. That way they will have higher chances of getting the latest fixes.

Reply:
I do not agree. I think, depending on what PRs are in the pipeline, we are working on 0.2.1 or 0.3.0. It could be the case that we wanted a preliminary release and did not want to wait for more unrelated patches. If we plan to release 0.2.1, we develop the version on master and merge it to branch 0.2. once the release is finished. Does it make sense?

I agree that depending on the changes and the range of versions we want to support, you have to backport to multiple minor versions.

Regarding the questions:
1) Yes, I think it is normal to have releases with flaws. Depending on the severity of the issues and whether the version is still used, we decide for or against backporting.
2) That is a good advice!

Reply:
I thought about your comments again and I agree. I changed the example to cover versions 0.2.0, 0.2.1 and 0.3.0.